### PR TITLE
Log run info decorator

### PR DESCRIPTION
**Pull request by bentsi**
**Added log_run_info decorator and minor bug fixes**
 About decorator (from doc string):

>         Decorator that prints BEGIN before the function runs and END when function finished running.
>         Uses function name as a name of action or string that can be given to the decorator.
>         If the function is a method of a class object, the class name will be printed out.
> 
>         Usage examples:
>             @log_run_info
>             def foo(x, y=1):
>                 pass
>             In: foo(1)
>             Out:
>                 BEGIN: foo
>                 END: foo (ran 0.000164)s
> 
>             @log_run_info("Execute nemesis")
>             def disrupt():
>                 pass
>             In: disrupt()
>             Out:
>                 BEGIN: Execute nemesis
>                 END: Execute nemesis (ran 0.000271)s

### CODE REVIEW

**Comment by bentsi** [comment]
@roydahan  @amoskong  @larisau 
Please review

File: sdcm/utils.py [hunk 1]
```diff
@@ -104,3 +105,53 @@ def inner(*args, **kwargs):
...
+            start_time = datetime.datetime.now()
+            logger.info("BEGIN: %s", action)
+            res = func(*args, **kwargs)
+            end_time = datetime.datetime.now()
+            logger.info("END: %s (ran %ss)", action, (end_time - start_time).total_seconds())
+            return res
+        return inner
+
+    if callable(arg):  # when decorator is used without a string message
+        return _inner(arg)
```
Comment (larisau) [inline sdcm/utils.py]: `return _inner(arg, arg.__name__) ` 
and the above msg_str is not needed

File: sdcm/utils.py [hunk 2]
```diff
@@ -104,3 +105,53 @@ def inner(*args, **kwargs):
...
+    def _inner(func, msg=None):
+        if msg:
+            msg_str = msg
+        else:
+            msg_str = func.__name__
+
+        @wraps(func)
+        def inner(*args, **kwargs):
+            class_name = ""
+            # import pdb; pdb.set_trace()
```
Comment (larisau) [inline sdcm/utils.py]: forgot debug

File: sdcm/utils.py [hunk 3]
```diff
@@ -16,8 +16,9 @@
 import os
 import glob
 
-import time
+import time, datetime
 from functools import wraps
+from inspect import ismethod
```
Comment (larisau) [inline sdcm/utils.py]: import not used?

Comment (bentsi) [reply sdcm/utils.py]: right, removing it

File: sdcm/utils.py (hunk 2 above)
Comment (bentsi) [reply sdcm/utils.py]: :+1:

**Comment by larisau** [comment]
Yes, I see, but in many cases it was used for debug time and left forever, and can be removed actually...
Anyway, it's not critical :)

File: sdcm/tester.py [hunk 4]
```diff
@@ -1014,6 +1014,12 @@ def clean_resources(self):
                 for node in self.db_cluster.nodes:
                     node.instance.stop()
                 self.db_cluster = None
+            elif self._failure_post_behavior == 'keep':
```
Comment (amoskong) [inline sdcm/tester.py]: good catch!

File: sdcm/utils.py [hunk 5]
```diff
@@ -104,3 +104,47 @@ def inner(*args, **kwargs):
...
+    """
+    def _inner(func, msg=None):
+        @wraps(func)
+        def inner(*args, **kwargs):
+            class_name = ""
+            if len(args) > 0 and func.__name__ in dir(args[0]):
+                class_name = " <%s>" % args[0].__class__.__name__
+            action = "%s%s" % (msg, class_name)
+            start_time = datetime.datetime.now()
+            logger.info("BEGIN: %s", action)
```
Comment (amoskong) [inline sdcm/utils.py]: original msgs are sent by self.log.debug(), here you switched it to INFO level. Is it expected?

File: sdcm/cluster.py [hunk 6]
```diff
@@ -786,7 +786,7 @@ def stop_task_threads(self, timeout=10):
     def destroy(self):
         raise NotImplementedError('Derived classes must implement destroy')
 
-    def wait_ssh_up(self, verbose=True, timeout=None):
+    def wait_ssh_up(self, verbose=True, timeout=300):
```
Comment (amoskong) [inline sdcm/cluster.py]: resonable.

**Pull request merged**